### PR TITLE
Optimized sel sorted algorithm

### PR DESCRIPTION
**Pull request by zhongwuzw**
Just calculate what we want.

### CODE REVIEW

**Comment by RingoD** [comment]
What do you expect to optimize? Performance?

**Comment by zhongwuzw** [comment]
@RingoD Yeah, At least two times faster than `NSNumericSearch`.

File: objc/source/interface/orm/binding/WCTBinding.mm [hunk 1]
```diff
@@ -28,6 +28,24 @@
 #import <WCDB/WCTPropertyMacro.h>
 #import <objc/runtime.h>
 
+static inline NSInteger getCounterPart(NSString *selName)
+{
+    WCTAssert(selName.length > 3, "selName's length must bigger than 3.");
```
Comment (zhongwuzw) [inline objc/source/interface/orm/binding/WCTBinding.mm]: No other edge check, because we use internal, we can control by myself.

**Comment by RingoD** [comment]
Do you have any benchmark about how much the performance can increase ?

I mean, since it only runs once per class and most of the classes will not have more than 100 even 20  methods, this optimization seems like unnecessary. 

If there is not a visually enhancement, I prefer to keep it simple.

**Comment by zhongwuzw** [comment]
The result like below. It's ok that you can prefer the original, I just only think `NSNumericSearch` on whole string is not necessary. 😂 

image

Simple benchmark:
```
        NSComparator comparator = ^NSComparisonResult(NSString *str1, NSString *str2) {
            NSInteger str1Integer = getCounterPart(str1);
            NSInteger str2Integer = getCounterPart(str2);
            
            if (str1Integer > str2Integer) {
                return NSOrderedDescending;
            } else if (str1Integer < str2Integer) {
                return NSOrderedAscending;
            } else {
                return NSOrderedSame;
            }
        };
        
        NSComparator comparator1 = ^NSComparisonResult(NSString *str1, NSString *str2) {
            return [str1 compare:str2 options:NSNumericSearch];
        };
        
        NSMutableArray<NSString *> *array = [NSMutableArray array];
        [array addObject:@"__wcdb_ORMColumnCoding_synthesize_13:"];
        [array addObject:@"__wcdb_ORMColumnCoding_synthesize_2222:"];
        [array addObject:@"__wcdb_ORMColumnCoding_synthesize_1121:"];
        [array addObject:@"__wcdb_ORMColumnCoding_synthesize_14:"];
        [array addObject:@"__wcdb_ORMColumnCoding_synthesize_10:"];
        [array addObject:@"__wcdb_ORMColumnCoding_synthesize_9:"];
        [array addObject:@"__wcdb_ORMColumnCoding_synthesize_8:"];
        
        NSDate *dateStart = [NSDate date];
        
        for (int i = 0; i < 1000000; i++) {
            [array sortedArrayUsingComparator:comparator1];
        }
        
        NSDate *dateEnd = [NSDate date];
        
        NSTimeInterval intervalFirst = [dateEnd timeIntervalSinceDate:dateStart];
        
        NSDate *dateStart1 = [NSDate date];
        
        for (int i = 0; i < 1000000; i++) {
            [array sortedArrayUsingComparator:comparator];
        }
        
        NSDate *dateEnd1 = [NSDate date];
        
        NSTimeInterval intervalSecond = [dateEnd1 timeIntervalSinceDate:dateStart1];
        
        NSLog(@"not opt is %f,opt is %f", intervalFirst, intervalSecond);
```

**Comment by RingoD** [comment]
Thanks for your code.

Since I already filter the non-WCDB-ORM methods, this optimization will be effective only developers define 100000 methods manually. 
So I still insist my opinion - the optimization should be done for those most costly code.

**Comment by RingoD** [comment]
No reply.

**Pull request closed without merging**